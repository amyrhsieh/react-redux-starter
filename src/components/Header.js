import React from 'react';
import { NavLink } from 'react-router-dom';

import styles from './header.scss';

const Header = () => (
  <header>
    <div>
      <h1>React Redux Demo</h1>
    </div>
    <div className="justify-content-center nav">
      <div className="nav-item">
        <NavLink to="/" exact className="nav-link" activeClassName="header__active">Home</NavLink>
      </div>
      <div className="nav-item">
        <NavLink to="/blog" className="nav-link" activeClassName="header__active">Blog</NavLink>
      </div>
      <div className="nav-item">
        <NavLink to="/events" className="nav-link" activeClassName="header__active">Events</NavLink>
      </div>
    </div>
  </header>
);

export default Header;
