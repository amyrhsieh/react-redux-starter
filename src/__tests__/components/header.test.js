import React from 'react';
import renderer from 'react-test-renderer';
import Header from 'Components/Header.js';
import { StaticRouter } from 'react-router';

describe('Header (Snapshot)', () => {
  it('Header component renders header nav', () => {
    const component = renderer.create(
      <StaticRouter>
        <Header />
      </StaticRouter>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
