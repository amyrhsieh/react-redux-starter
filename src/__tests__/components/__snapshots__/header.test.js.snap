// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header (Snapshot) Header component renders header nav 1`] = `
<header>
  <div
    activeKey="/home"
    className="justify-content-center nav"
  >
    <div
      className="nav-item"
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/events"
        onClick={[Function]}
      >
        Events
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/blog"
        onClick={[Function]}
      >
        Blog
      </a>
    </div>
  </div>
</header>
`;
